src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Logging/AbstractLoggerFactory.AggregateLogger.cs

```diff
@@ -8,15 +8,35 @@ namespace Microsoft.CodeAnalysis.Razor.Logging;
 
 internal abstract partial class AbstractLoggerFactory
 {
-    private class AggregateLogger(ImmutableArray<ILogger> loggers) : ILogger
+    private sealed class LazyLogger(Lazy<ILoggerProvider, LoggerProviderMetadata> lazyProvider, string categoryName)
     {
-        private ImmutableArray<ILogger> _loggers = loggers;
+        private readonly LoggerProviderMetadata _metadata = lazyProvider.Metadata;
+        private readonly Lazy<ILogger> _lazyLogger = new(() => lazyProvider.Value.CreateLogger(categoryName));
+
+        public ILogger Instance => _lazyLogger.Value;
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            // If the ILoggerProvider's metadata has a minimum log level, we can use that
+            // rather than forcing the ILoggerProvider to be created.
+            if (_metadata.MinimumLogLevel is LogLevel minimumLogLevel)
+            {
+                return logLevel.IsAtLeast(minimumLogLevel);
```

Comment:
Feels like a violation of contract to not also call `Instance.IsEnabled` if this check passes. Perhaps this should change to `if (!logLevel.IsAtLeast...) return false;`?

Reply:
Derp! Yes! Thank you.

Now close your eyes and move along. Nothing to see here. :smile: